cvat-ui/src/components/md-guide/guide-page.tsx

```diff
@@ -0,0 +1,191 @@
+    const id = +useParams<{ id: string }>().id;
+    const [guide, setGuide] = useState<AnnotationGuide>(
+        new AnnotationGuide({
+            ...(instanceType === 'project' ? { project_id: id } : { task_id: id }),
+            markdown: value,
+        }),
+    );
+    const [fetching, setFetching] = useState(false);
+
+    useEffect(() => {
+        setFetching(true);
+        const promise = instanceType === 'project' ? core.projects.get({ id }) : core.tasks.get({ id });
+        promise.then(([instance]: [Task | Project]) => (
+            instance.guide()
+        )).then((guideInstance: AnnotationGuide | null) => {
+            if (guideInstance) {
+                return (async () => guideInstance)();
+            }
+            return guide.save();
+        }).then((guideInstance: AnnotationGuide) => {
+            if (isMounted()) {
+                setValue(guideInstance.markdown);
+                setGuide(guideInstance);
+            }
+        }).catch((error: any) => {
+            if (isMounted()) {
+                notification.error({
+                    message: `Could not receive guide for the ${instanceType} ${id}`,
+                    description: error.toString(),
+                });
+            }
+        }).finally(() => {
+            if (isMounted()) {
+                setFetching(false);
+            }
+        });
+    }, []);
+
+    const submit = useCallback((currentValue: string) => {
+        guide.markdown = currentValue;
+        setFetching(true);
+        guide.save().then((result: AnnotationGuide) => {
+            if (isMounted()) {
+                setValue(result.markdown);
+                setGuide(result);
+            }
+        }).catch((error: any) => {
+            if (isMounted()) {
+                notification.error({
+                    message: 'Could not save guide on the server',
+                    description: error.toString(),
+                });
+            }
+        }).finally(() => {
+            if (isMounted()) {
+                setFetching(false);
+            }
+        });
+    }, [guide, fetching]);
+
+    const handleInsert = async (event: React.ClipboardEvent | React.DragEvent, files: FileList): Promise<void> => {
+        if (files.length && guide.id) {
+            event.preventDefault();
+            const assetsToAdd = Array.from(files);
+            const addedAssets: [File, string][] = [];
+
+            if (mdEditorRef.current) {
+                const { textArea } = mdEditorRef.current.commandOrchestrator;
+                const { selectionStart, selectionEnd } = textArea;
+                const computeNewValue = (): string => {
+                    const addedStrings = addedAssets.map(([file, uuid]) => {
+                        if (file.type.startsWith('image/')) {
+                            return (`![image](/api/assets/${uuid})`);
+                        }
+                        return (`[${file.name}](/api/assets/${uuid})`);
```

Comment:
We can add any file type? Is that expected?

Reply:
On client yes, file types are handled by the server. 
I did it this way to avoid dublicating and synchronization